nemo/collections/common/callbacks/callbacks.py

```diff
@@ -30,3 +33,67 @@ def on_epoch_end(self, trainer, pl_module):
         curr_time = time.time()
         duration = curr_time - self.epoch_start
         trainer.logger.log_metrics({"epoch_time": duration}, step=trainer.global_step)
+
+
+class MachineTranslationLogEvalCallback(Callback):
```

Comment:
It looks to me that you are implementing BLEU score calculation using callback. 
A prefer way of calculating metrics with PTL is by implementing a Metric class. Let's discuss if we can implement this using Metric object similarly how it is done for WER calculation.